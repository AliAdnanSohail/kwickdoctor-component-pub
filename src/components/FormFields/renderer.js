import React from 'react';
import classnames from 'classnames';

import styles, { error as errorStyles, label as labelStyles } from './styles';

export default render => ({
  input, meta, label, ...rest
}) => (
  <div className={classnames('form-field', rest.containerClassName)}>
    <label htmlFor={rest.id}>{label}</label>
    {render(input, label, {
      ...rest,
      className: classnames({
        invalid: meta && meta.error && meta.touched,
        active: meta && meta.active,
      }),
    })}
    {meta && meta.error && meta.touched ? <div className="error">{meta.error}</div> : undefined}

    <style jsx>{styles}</style>
    <style jsx>{labelStyles}</style>
    <style jsx>{errorStyles}</style>
  </div>
);
