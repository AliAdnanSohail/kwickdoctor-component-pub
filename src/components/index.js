export { default as AvatarPicker } from './AvatarPicker';
export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as CardPicker } from './CardPicker';
export { default as DatePicker } from './DatePicker';
export { default as FileInput } from './FileInput';
export { default as Form } from './Form';
export { default as Modal } from './Modal';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';

export * from './FormFields';
export * from './Stepper';
