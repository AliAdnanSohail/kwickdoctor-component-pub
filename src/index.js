export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Calendar } from './components/Calendar';
export { default as Card } from './components/Card';
export { default as Datepicker } from './components/Datepicker';
export { default as TextInput } from './components/TextInput';
export { default as Toast } from './components/Toast';
<<<<<<< HEAD
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as AvatarPicker } from './components/AvatarPicker';
=======
>>>>>>> master
