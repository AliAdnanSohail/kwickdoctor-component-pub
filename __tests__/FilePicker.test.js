import React, { Component } from 'react';
import renderer from 'react-test-renderer';

import { FilePicker } from '../src';

class FilePickerWrapper extends Component {
  constructor(props) {
    super(props);

    this.state = { fileName: undefined };
  }

  handleChange = (file, fileName) => {
    this.setState({ fileName });
    this.props.onChange(file, fileName);
  };

  render() {
    const { id, name, placeholder } = this.props;
    const { fileName } = this.state;

    return (
      <FilePicker
        id={id}
        fileName={fileName}
        name={name}
        placeholder={placeholder}
        onChange={this.handleChange}
      />
    );
  }
}

it('renders correctly', () => {
  const element = renderer
    .create(<FilePickerWrapper id="file-input" name="name" placeholder="Upload license" />)
    .toJSON();

  expect(element).toMatchSnapshot();
});
