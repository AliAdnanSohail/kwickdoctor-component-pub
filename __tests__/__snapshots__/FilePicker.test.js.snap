// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<label
  className="jsx-1336855819 jsx-2881818901 upload-file upload-file--selected"
  htmlFor="file-picker"
>
  <div
    className="jsx-1336855819 jsx-2881818901 upload-file__label-container"
  >
    <svg
      aria-label="upload"
      className="grommetux-control-icon grommetux-control-icon-upload grommetux-control-icon--responsive"
      height="24px"
      role="img"
      version="1.1"
      viewBox="0 0 24 24"
      width="24px"
    >
      <path
        d="M1,17 L1,23 L23,23 L23,17 M12,2 L12,19 M5,9 L12,2 L19,9"
        fill="none"
        stroke="#000"
        strokeWidth="2"
      />
    </svg>
    <div
      className="jsx-1336855819 jsx-2881818901 upload-file__label"
    >
      file-picker
    </div>
  </div>
  <input
    className="jsx-1336855819 jsx-2881818901 input-file"
    id="file-picker"
    onBlur={[Function]}
    onChange={[Function]}
    type="file"
  />
</label>
`;
